### PR TITLE
check usage of Layout measurePolicy in sample

### PR DESCRIPTION
**Pull request by dima-avdeev-jb**
Check crash, what fixed in PR https://github.com/JetBrains/androidx/pull/237
Add **gradle task runLayout** with desktop sample of usage Layout and MeasurePolicy.
In sample different views iterates with delay in 1 second. So we can check all of them.
Crash exception not exists.

### CODE REVIEW

File: compose/desktop/desktop/samples/src/jvmMain/kotlin/androidx/compose/desktop/examples/layout/Main.jvm.kt [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import androidx.compose.runtime.*
+import androidx.compose.ui.awt.ComposeDialog
+import androidx.compose.ui.window.Dialog
+import androidx.compose.ui.window.launchApplication
+import java.awt.Dimension
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.delay
+
+fun main() {
+    MainScope().launchApplication {
```
Comment (igordmn) [inline compose/desktop/desktop/samples/src/jvmMain/kotlin/androidx/compose/desktop/examples/layout/Main.jvm.kt]: The idiomatic API to open dialogs is:
```
fun main() = application {
   // resizable is true by default
  Dialog(
     state = rememberDialogState(width = 400,. height = 600),
     undecorated = true
  ) {

  }
}
```
And why Dialog, not Window, which can be created by singleWindowApplication?

File: compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/LayoutWithWorkaround.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import androidx.compose.ui.layout.MeasureScope
+import androidx.compose.ui.unit.Constraints
+
+/*
+ * TODO: This is a version of Layout not using SAM-converted [MeasurePolicy] interface.
+ * Remove me, when SAM conversion for [MeasurePolicy] works again in Kotlin 1.7.0
+ * Check with Desktop jvm test DesktopAlertDialogTest.alignedToCenter_inPureWindow
+ */
+@Composable
+internal inline fun LayoutWithWorkaround(
```
Comment (igordmn) [inline compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/LayoutWithWorkaround.kt]: As we moved this into another PR, we need to remove it from this PR

File: compose/ui/ui/src/desktopTest/kotlin/androidx/compose/ui/window/dialog/DialogTest.kt [hunk 3]
```diff
@@ -65,7 +65,8 @@ class DialogTest {
 
             fun createWindow() = ComposeDialog().apply {
                 size = Dimension(300, 200)
-
+                isResizable = true
+                isUndecorated = true
```
Comment (igordmn) [inline compose/ui/ui/src/desktopTest/kotlin/androidx/compose/ui/window/dialog/DialogTest.kt:69]: Do we need this change in DialogTest?

File: compose/desktop/desktop/samples/src/jvmMain/kotlin/androidx/compose/desktop/examples/layout/Main.jvm.kt (hunk 1 above)
Comment (dima-avdeev-jb) [reply compose/desktop/desktop/samples/src/jvmMain/kotlin/androidx/compose/desktop/examples/layout/Main.jvm.kt]: Because I want to check this line of code https://github.com/JetBrains/androidx/blob/e529c7e79bee073a21bc789f2dc5b10b60e4dddd/compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/window/UndecoratedWindowResizer.desktop.kt#L69

It is available only with custom creation of ComposeDialog with flags isResizable and isUndecorated:
ComposeDialog().apply {
            isResizable = true
            isUndecorated = true
        }

File: compose/ui/ui/src/desktopTest/kotlin/androidx/compose/ui/window/dialog/DialogTest.kt (hunk 3 above)
Comment (dima-avdeev-jb) [reply compose/ui/ui/src/desktopTest/kotlin/androidx/compose/ui/window/dialog/DialogTest.kt:69]: It covers this line of code [androidx/compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/window/UndecoratedWindowResizer.desktop.kt](https://github.com/JetBrains/androidx/blob/e529c7e79bee073a21bc789f2dc5b10b60e4dddd/compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/window/UndecoratedWindowResizer.desktop.kt#L69)

Comment (dima-avdeev-jb) [reply compose/ui/ui/src/desktopTest/kotlin/androidx/compose/ui/window/dialog/DialogTest.kt:69]: So we can catch in unit test potential bugs

File: compose/desktop/desktop/samples/src/jvmMain/kotlin/androidx/compose/desktop/examples/layout/Main.jvm.kt (hunk 1 above)
Comment (igordmn) [reply compose/desktop/desktop/samples/src/jvmMain/kotlin/androidx/compose/desktop/examples/layout/Main.jvm.kt]: > It is available only with custom creation of ComposeDialog

`Dialog` calls the same code under the hood, and creates UndecoratedWindowResizer

Comment (igordmn) [reply compose/desktop/desktop/samples/src/jvmMain/kotlin/androidx/compose/desktop/examples/layout/Main.jvm.kt]: `Window` also calls this code

File: compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/LayoutWithWorkaround.kt (hunk 2 above)
Comment (dima-avdeev-jb) [reply compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/LayoutWithWorkaround.kt]: Yeah, correct. I will remove it.

File: compose/ui/ui/src/desktopTest/kotlin/androidx/compose/ui/window/dialog/DialogTest.kt (hunk 3 above)
Comment (igordmn) [reply compose/ui/ui/src/desktopTest/kotlin/androidx/compose/ui/window/dialog/DialogTest.kt:69]: Ack

File: compose/desktop/desktop/samples/src/jvmMain/kotlin/androidx/compose/desktop/examples/layout/Main.jvm.kt (hunk 1 above)
Comment (dima-avdeev-jb) [reply compose/desktop/desktop/samples/src/jvmMain/kotlin/androidx/compose/desktop/examples/layout/Main.jvm.kt]: Ok, good suggestion. I will change it.

File: compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/LayoutWithWorkaround.kt (hunk 2 above)
Comment (dima-avdeev-jb) [reply compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/LayoutWithWorkaround.kt]: Done

File: compose/desktop/desktop/samples/src/jvmMain/kotlin/androidx/compose/desktop/examples/layout/Main.jvm.kt (hunk 1 above)
Comment (dima-avdeev-jb) [reply compose/desktop/desktop/samples/src/jvmMain/kotlin/androidx/compose/desktop/examples/layout/Main.jvm.kt]: Done

**Review by igordmn — APPROVED** [review]
